### PR TITLE
[SYCL] Fix zstd build when static zstd libraries are not found

### PR DESCRIPTION
**Pull request by uditagarwal97**
**Problem:**

When we use `LLVM_ENABLE_ZSTD=ON` along with `LLVM_USE_STATIC_ZSTD=ON`, LLVM config fails with the following error when libzstd.a is not found but libzstd.so is found:
```
CMake Error at lib/Support/CMakeLists.txt:327 (get_property):
  get_property could not find TARGET zstd::libzstd_static.  Perhaps it has
  not yet been created.


CMake Error at lib/Support/CMakeLists.txt:330 (get_property):
  get_property could not find TARGET zstd::libzstd_static.  Perhaps it has
  not yet been created.
```
The bug is in upstream and in the following lines: https://github.com/intel/llvm/blob/sycl/llvm/lib/Support/CMakeLists.txt#L32 - In the `else()`, the code just assumes that static libraries are available.

**Solution:**

When user specifies `LLVM_USE_STATIC_ZSTD=ON`, we should enable ZSTD only if static zstd library is present. Otherwise, the build should just proceed normally since zstd is optional, even in upstream.

### CODE REVIEW

File: llvm/cmake/config-ix.cmake [hunk 1]
```diff
@@ -157,7 +157,14 @@ if(LLVM_ENABLE_ZSTD)
     find_package(zstd QUIET)
   endif()
 endif()
+
+# If LLVM_USE_STATIC_ZSTD is specified, make sure we enable zstd only if static
+# libraries are found.
```
Comment (uditagarwal97) [inline llvm/cmake/config-ix.cmake:162]: I'll create a bug report and PR to upstream this change, if approved.

File: llvm/cmake/config-ix.cmake [hunk 2]
```diff
@@ -157,7 +157,14 @@ if(LLVM_ENABLE_ZSTD)
...
+if(LLVM_USE_STATIC_ZSTD AND NOT TARGET zstd::libzstd_static)
```
Comment (sarnex) [inline llvm/cmake/config-ix.cmake:163]: sorry what is `zstd::libzstd_static` and who sets it? it is set by find_package if it finds the static lib?

Comment (uditagarwal97) [reply llvm/cmake/config-ix.cmake:163]: Yes, `zstd::libzstd_static` is set during find_package if it finds static ZSTD libraries. https://github.com/intel/llvm/blob/sycl/llvm/cmake/modules/Findzstd.cmake#L60

**Review by sarnex — APPROVED** [review]
makes sense to me, but lets also submit to upstream like you said

File: llvm/cmake/config-ix.cmake (hunk 1 above)
Comment (uditagarwal97) [reply llvm/cmake/config-ix.cmake:162]: For upstreaming:
issue: https://github.com/llvm/llvm-project/issues/113583
PR: https://github.com/llvm/llvm-project/pull/113584

**Comment by uditagarwal97** [comment]
@intel/llvm-gatekeepers the PR is ready to be merged.

**Pull request merged**